src/Shared/TaskParameter.cs

```diff
@@ -1003,12 +1003,7 @@ private IEnumerable<KeyValuePair<string, string>> EnumerateMetadataEager()
             {
                 if (_customEscapedMetadata == null || _customEscapedMetadata.Count == 0)
                 {
-#if TASKHOST
```

Comment:
Getting rid of ifdefs? 🙌🏻

Reply:
Oh how I wish that were true 😭

At least we can let the compiler deal with it!